src/lib/y2storage/filesystems/base.rb

```diff
@@ -55,6 +56,29 @@ class Base < Mountable
       #   @return [SpaceInfo]
       storage_forward :detect_space_info, as: "SpaceInfo"
 
+      # smarted space info including caching and handling not probed devices.
+      #   @return [SpaceInfo]
+      def space_info
```

Comment:
As commented on IRC. If we are only interested right now in knowing the free space, I would only implement a `#free_size` or `#free_space` method. Handcrafting a whole `SpaceInfo` object implies we should adapt this method in the future every time a new attribute is added to `SpaceInfo` in libstorage-ng. And maybe we don't have a good way to infer such new attribute.

So it's better to not over-promise.

Reply:
so we just return free_size if detect pass and estimation if detect failed, right?